provider/datagen/src/transform/cldr/cldr_serde/relativetime.rs

```diff
@@ -0,0 +1,146 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Serde structs representing CLDR JSON numbers.json files.
+//!
+//! Sample file:
+//! <https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-dates-full/main/en/dateFields.json>
+
+use std::collections::HashMap;
+
+use icu_locid::LanguageIdentifier;
+use serde::{
+    de::{Error, IgnoredAny, Visitor},
+    Deserialize,
+};
+
+#[derive(Debug, Deserialize)]
+pub struct PluralRulesPattern {
+    #[serde(rename = "relativeTimePattern-count-zero")]
+    pub zero: Option<String>,
+    #[serde(rename = "relativeTimePattern-count-one")]
+    pub one: Option<String>,
+    #[serde(rename = "relativeTimePattern-count-two")]
+    pub two: Option<String>,
+    #[serde(rename = "relativeTimePattern-count-few")]
+    pub few: Option<String>,
+    #[serde(rename = "relativeTimePattern-count-many")]
+    pub many: Option<String>,
+    #[serde(rename = "relativeTimePattern-count-other")]
+    pub other: Option<String>,
+}
+
+#[derive(Debug)]
+pub struct Relative {
+    pub count: i8,
+    pub pattern: String,
+}
+
+#[derive(Debug)]
+pub struct Field {
+    pub display_name: Option<String>,
+    pub relative_period: Option<String>,
+    pub relatives: Vec<Relative>,
+    pub past: Option<PluralRulesPattern>,
+    pub future: Option<PluralRulesPattern>,
+}
+
+pub struct FieldVisitor;
```

Comment:
Praise: This looks like a good use of a visitor, to clean up a really ugly CLDR JSON representation into a cleaner Rust one.